example/sparse/linear_classification.py

```diff
@@ -56,19 +81,34 @@
     kvstore = args.kvstore
     batch_size = args.batch_size
     optimizer = args.optimizer
-
+    log_interval = args.log_interval
+    lr = args.lr
     # create kvstore
     kv = mx.kvstore.create(kvstore) if kvstore else None
     rank = kv.rank if kv else 0
     num_worker = kv.num_workers if kv else 1
 
     # dataset
-    num_features = AVAZU['num_features']
-    data_dir = os.path.join(os.getcwd(), 'data')
-    train_data = os.path.join(data_dir, AVAZU['train'])
-    val_data = os.path.join(data_dir, AVAZU['test'])
-    get_libsvm_data(data_dir, AVAZU['train'], AVAZU['url'])
-    get_libsvm_data(data_dir, AVAZU['test'], AVAZU['url'])
+    if args.model == 'linear':
+        num_features = AVAZU['num_features']
+        data_dir = os.path.join(os.getcwd(), 'data')
+        train_data = os.path.join(data_dir, AVAZU['train'])
+        val_data = os.path.join(data_dir, AVAZU['test'])
+        get_libsvm_data(data_dir, AVAZU['train'], AVAZU['url'])
+        get_libsvm_data(data_dir, AVAZU['test'], AVAZU['url'])
+        # The positive class weight, says how much more we should upweight the importance of
+        # positive instances in the objective function.
+        # This is used to combat the extreme class imbalance.
+        model = linear_model(AVAZU['num_features'], AVAZU['positive_class_weight'])
+    else:
+        num_features = ADULT['num_features']
```

Comment:
I think this example should be moved to another file since it's not a linear model anymore.

Reply:
linear classification and wide & deep classification are separated to two files now.